### PR TITLE
pvalue correction in analyse groups

### PR DESCRIPTION
**Pull request by AgnesBaud**
## Description

Optional correction of p-values with multipletests from statsmodels.stats.multitest.
Different pvalues output = series or dataframe, symmetric or not

#### Changelogs

* Correction of p-values in analyse_groups (diversity)

## Definition of Done

* [ ]  New code is tested with unit tests
* [ ]  Documentation has been updated
* [ ]  Pull Request has been revised by a maintainer of the project

### CODE REVIEW

**Comment by AgnesBaud** [comment]
Missing tests

File: moonstone/analysis/diversity/base.py [hunk 1]
```diff
@@ -168,12 +172,37 @@ def analyse_groups(
...
+        :param output_pvalue: {'series', 'dataframe'}
+        :param sym: whether generated dataframe (or MultiIndexed series) is symetric or half-full
         """
         filtered_metadata_df = self._get_filtered_df_from_metadata(metadata_df)
         df = self._get_grouped_df(filtered_metadata_df[group_col])
-        pval = statistical_test_groups_comparison(
-            df[self.DIVERSITY_INDEXES_NAME], df[group_col], stats_test
-        )
+
+        if correction_method is not None:
```
Comment (khillion) [inline moonstone/analysis/diversity/base.py:185]: What you can do here to make the code easier to read and avoid long method is to refactor all of your code into a method of the class:

```python
df = self._get_grouped_df(filtered_metadata_df[group_col])
```

here you would get roughly something like:

```python
pval = self._run_statistical_test_groups(stats_test, correction_method, sym, output_pvalue)
```

and have you method

```python
def self _run_statistical_test_groups(self, stats_test: str, correction_method: str, sym: bool, output_pvalue: str):
    #your code
```

Therefore you only give what's necessary to perform the action and it gets much easier to debug but also test.

File: moonstone/analysis/statistical_test.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def statistical_test_groups_comparison(series: pd.Series, group_series: pd.Serie
                                        output: str = 'dataframe', sym: str = True, **kwargs):
     """
     :param output: {'series', 'dataframe'}
-    :param mirror: whether generated dataframe (matrix) (or MultiIndexed series) is symetric or half-full
+    :param sym: whether generated dataframe (or MultiIndexed series) is symetric or half-full
```
Comment (khillion) [inline moonstone/analysis/statistical_test.py:41]: make me think that maybe you could add tests in test_statistical_test.py that handles both cases?

**Review by khillion — COMMENTED** [review]
minor comment, tiny bit of refactoring and it will be good from my side :)

**Review by khillion — APPROVED** [review]
looks good to me! merge whenever you want!

**Pull request merged**